x-pack/platform/plugins/shared/dashboard_agent/server/tools/utils.ts

```diff
@@ -32,12 +43,35 @@ export const checkDashboardToolsAvailability = async ({
  * This is a temporary function to handle lens API schema conversion.
  */
 export const normalizePanels = (panels: unknown[] | undefined): DashboardPanel[] => {
-  return (panels ?? []).map((panel, index) => {
-    return buildLensPanelFromApi(panel as LensApiSchemaType, index);
-  });
+  const panelConfigs = panels ?? [];
+  const dashboardPanels: DashboardPanel[] = [];
+  let currentX = 0;
+  let currentY = 0;
+
+  for (const panel of panelConfigs) {
+    const config = panel as LensApiSchemaType;
```

Comment:
Why is the type cast needed here? Is there a way to get the correct type without having to do this?